src/main/java/io/jenkins/plugins/audit/config/AuditLogConfiguration.java

```diff
@@ -0,0 +1,44 @@
+package io.jenkins.plugins.audit.config;
+
+import hudson.Extension;
+import jenkins.model.GlobalConfiguration;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import java.util.Objects;
+
+@Extension
+public class AuditLogConfiguration extends GlobalConfiguration {
+    private String logDestination;
+
+    public AuditLogConfiguration() {
+        super();
+        load();
+    }
+
+    public String getLogDestination() {
+        return logDestination;
+    }
+
+    @DataBoundSetter
+    public void setLogDestination(String logDestination) {
+        this.logDestination = logDestination;
+        save();
+        System.setProperty("auditFileName", logDestination);
```

Comment:
I'd suggest extracting this line and the one below into its own method. You'll want to execute it in the constructor after calling `load()`, too.

Reply:
Sure. If I call it in the constructor, the log destination may not set that time. Should we set the default here as well along with doing it in log4j2.xml file ?

Reply:
Please correct me if my understanding is not right

Reply:
So in that case, we may need to move the setting of system property out of that function as we will not have the logDestination defined to be set during the time of class instantiation. Am I making sense?

I tried adding it and it results to nullPointerException and so I commented it for now.

Reply:
Oh, right, we should only set the property if `logDestination` is not null. That will allow the default value from log4j2.xml to take over since the system property won't be defined. You could also add `System.removeProperty("logDestination")` if it's null, too.